backend/server/api/controllers/todo/controller.js

```diff
@@ -37,6 +38,35 @@ export class Controller {
       console.log(error);
     }
   }
+
+  async search(req, res) {
+    const searchTodo = {
+      useName: req.body.useName,
+      useDescription: req.body.useDescription,
+      useIsDone: req.body.useIsDone,
+      useDueDate: req.body.useDueDate,
+      name: req.body.name,
+      description: req.body.description,
+      isDone: req.body.isDone,
+      dueDate: req.params.dueDate,
```

Comment:
Why the `use*`? I think they're extraneous.

Reply:
It is to know whether to use any of the filters. At first I though to compare the property to an empty string to know if that search filter was used, but I stumbled with two problems:

1. It is allowed to create todos with an empty description and due date, which means that empty is now a valid value on those cases.
2. isDone is boolean, so I don't have any other value left on that variable to know if it is being used as a filter or not (since both true and false are valid values).